diff -urw ADIOS2-2.7.1.orig/source/h5vol/CMakeLists.txt ADIOS2-2.7.1/source/h5vol/CMakeLists.txt
--- ADIOS2-2.7.1.orig/source/h5vol/CMakeLists.txt	2021-02-09 09:04:38.000000000 -0600
+++ ADIOS2-2.7.1/source/h5vol/CMakeLists.txt	2021-10-20 12:37:00.771803602 -0500
@@ -3,7 +3,7 @@
 # accompanying file Copyright.txt for details.
 #------------------------------------------------------------------------------#
 
-add_library(adios2_h5vol SHARED
+add_library(adios2_h5vol
   H5VolReadWrite.c
   H5Vol.c
   H5Vol_attr.c
